added idempotency key support & bank update method

glassw opened the pull request:
- use special $key parameter to set the Idempotency-Key header
- bank update method allows to deactivate a bank account (similar to card)

godspeedelbow commented:
Hi, thanks for contributing! Just got back from holiday, hence belated reply.

I think `$key` is a pretty generic variable name which might confuse people, maybe change it to `$idempotencyKey`? 

Could you also add the following, pretty please? :)
- documentation to the README.md
- unit test, see test scripts - it's pretty straightforward.

Cheers!

glassw commented:
Hello Eelke,

Sorry for the late reply. I renamed the variable as per your suggestion.
Also updated the README and commented out the "engines" section in
package.json to avoid warnings in newer node versions.

As for the tests, didn't add those to the lib - have tested them in my own
project though. It seems several tests are still missing (e.g., $query,
bank create, ...).

Best regards



On Wed, Sep 7, 2016 at 2:05 PM, Eelke Boezeman <notifications@github.com>
wrote:

> Hi, thanks for contributing! Just got back from holiday, hence belated
> reply.
>
> I think $key is a pretty generic variable name which might confuse
> people, maybe change it to $idempotencyKey?
>
> Could you also add the following, pretty please? :)
>
>    - documentation to the README.md
>    - unit test, see test scripts - it's pretty straightforward.
>
> Cheers!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/larafale/mangopay/pull/73#issuecomment-245259676>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ANFMXvlS-VKSJ6fQwTEXztL0ywB00pzrks5qnqiHgaJpZM4Jwhbp>
> .
>



-- 
http://glasswair.com

godspeedelbow commented on package.json:
```diff
@@ -9,9 +9,6 @@
     "type": "git",
     "url": "git://github.com/larafale/mangopay.git"
   },
-  // "engines": {
```
better to remove it all together, not valid JSON like this :)

glassw replied:
Ow, fixed that a few days ago, commit 63c8208 should be fine ;)

godspeedelbow commented:
@glassw 1.3.0 is published to npm with idempotency support
